altimeter_ms5607_spi/altimeter_ms5607_spi.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#if defined(STM32F1)
+#include "stm32f1xx_hal.h"
+#elif defined(STM32F4xx)
+#include "stm32f4xx_hal.h"
+#endif
+
+#ifndef SERIAL_TIMEOUT
+#define SERIAL_TIMEOUT 10
```

Comment:
IMO this would make more sense as a class member with a default value instead of a macro. That way, it's easier to configure and there's no chance we use the same macro name for different drivers, unless that's the intention with this.

Reply:
I've used the same macro in my memory and IMU drivers, and I think we'll use "10" for all of the protocol timeout. I get the point that it is easier to configure, but for an end user who simply wants to initialise a class and then read values, having it as a macro that main.cc doesn't see is better in my opinion

Reply:
For a user that doesn't want to figure it out, having a default value solves this problem. Having a macro with the same name for everything has the potential to screw us up if, for example, we are having problems with the altimeter responding slowly so we increase the timeout, but this inadvertently slows down the IMU and memory when we didn't want to.

Reply:
Having a shared serial timeout for all drivers was the original intention, but it doesn't actually make a whole lot of sense given each device operates at s different speed, will change so it's part of the class

Reply:
we could also make the default for all drivers configurable by macro